gen-ai-rag-vertex-ai-vector-search/ingestion.tf

```diff
@@ -0,0 +1,78 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Aspects to setup ingestion from Cloud Storage bucket to processing job.
+
+resource "google_storage_bucket" "ingest" {
+  name     = "ingest-${local.unique_str}"
+  location = var.region
+}
+
+
+resource "google_pubsub_topic" "ingest" {
+  name = "ingest-${local.unique_str}"
+
+  ingestion_data_source_settings {
+    cloud_storage {
+      bucket = google_storage_bucket.ingest.name
+    }
+  }
+}
+
+# Allow the Pub/Sub service account the ability to publish messages
+resource "google_project_iam_member" "pubsub" {
+  project = var.project_id
+  role    = "roles/pubsub.publisher"
+  member  = "serviceAccount:service-${data.google_project.default.project_id}@gcp-sa-pubsub.iam.gserviceaccount.com"
+}
+
+
+# Allow the Pub/Sub service account permissions to access the bucket
+resource "google_storage_bucket_iam_member" "pubsub" {
+  bucket = google_storage_bucket.ingest.name
+  role   = "roles/storage.admin"
```

Comment:
what kind of access is needed here? access the bucket reads to me like viewer, but I'm just wondering if there are more specific privs that would be right here 

https://cloud.google.com/storage/docs/access-control/iam-roles

Reply:
Based on the `terraform apply` error messages, the permissions need to include both `storage.buckets.get` and `storage.objects.get`. There is no role in a standard Google Cloud project that has both of these permissions except `roles/storage.admin`, _within the scope of Cloud Storage_. 

(There are other roles, but these are from different APIs, e.g. Composer Worker or Cloud Run Source Deployer, but they also come with tens or hundreds of other permissions.)

This is a similar issue to what terraform-cloud-client-api had, which used [a custom role](https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/blob/main/infra/main.tf#L150-L157) as a work around.

Reply:
Based on updated strategy, this IAM isn't required